File: AIML Project - Server/test_mongo_connection.js
require("dotenv").config();
const mongoose = require("mongoose");
async function testConnection() {
  console.log("=".repeat(50));
  console.log("Testing MongoDB Connection from Node Server");
  console.log("=".repeat(50));
  const mongoUri = process.env.MONGO_URI;
  console.log(`\n📍 MongoDB URI: ${mongoUri.substring(0, 30)}...`);
  try {
    console.log("\n🔄 Attempting to connect...");
    const conn = await mongoose.connect(mongoUri);
    console.log(`✅ Connection successful!`);
    console.log(`🌐 Host: ${conn.connection.host}`);
    console.log(`📦 Database: ${conn.connection.name}`);
    const collections = await conn.connection.db.listCollections().toArray();
    const salesExists = collections.some((col) => col.name === "sales");
    if (salesExists) {
      const count = await conn.connection.db.collection("sales").countDocuments();
      console.log(`📊 Sales collection found with ${count} documents`);
    } else {
      console.log("⚠️  Sales collection not found");
      console.log("💡 Run migrate_csv_to_mongo.py from ML service to populate data");
    }
    await mongoose.connection.close();
    console.log("\n✅ Test completed successfully");
  } catch (error) {
    console.error(`❌ Connection failed: ${error.message}`);
    console.log("\n💡 Troubleshooting:");
    console.log("   1. Check if MONGO_URI is set in .env file");
    console.log("   2. Verify MongoDB Atlas credentials");
    console.log("   3. Check network connectivity");
    console.log("   4. Ensure IP address is whitelisted in MongoDB Atlas");
  }
  process.exit(0);
}
testConnection();

File: AIML Project - ML Model/test_mongo_connection.py
"""
Test script to verify MongoDB connection from ML service.
Run this to ensure the ML model can connect to MongoDB Atlas.
"""
from preprocessing.load_data import load_sales_data
import os
from dotenv import load_dotenv
load_dotenv()
def test_connection():
    print("=" * 50)
    print("Testing MongoDB Connection from ML Service")
    print("=" * 50)
    mongo_uri = os.getenv("MONGO_URI", "Not found")
    print(f"\n📍 MongoDB URI: {mongo_uri[:30]}...")
    try:
        print("\n🔄 Attempting to load sales data...")
        df = load_sales_data()
        if df.empty:
            print("⚠️  Connection successful but no data found in 'sales' collection")
            print("💡 Run migrate_csv_to_mongo.py to populate the database")
        else:
            print(f"✅ Connection successful!")
            print(f"📊 Loaded {len(df)} records from MongoDB")
            print(f"\n📋 Columns: {list(df.columns)}")
            print(f"\n🔍 Sample data:")
            print(df.head(3))
    except Exception as e:
        print(f"❌ Connection failed: {str(e)}")
        print("\n💡 Troubleshooting:")
        print("   1. Check if MONGO_URI is set in .env file")
        print("   2. Verify MongoDB Atlas credentials")
        print("   3. Check network connectivity")
        print("   4. Ensure IP address is whitelisted in MongoDB Atlas")
if __name__ == "__main__":
    test_connection()